### PR TITLE
Enable netcoreapp2.1

### PR DESCRIPTION
**Pull request by eddynaka**
## Changes
- enable netcoreapp2.1 for all test projects that are possible
- added new TestApp for netcoreapp2.1

## References:
- How to add multiple dotnet versions in linux: https://github.com/actions/setup-dotnet/issues/25

### Checklist
- [x] I ran Unit Tests locally.

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public API reviewed

The PR will automatically request reviews from [code owners](https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/CODEOWNERS#L15) and trigger CI build and tests.

### CODE REVIEW

**Comment by eddynaka** [comment]
closing and reopening. we have one test that keeps working/not working from time to time.

File: .github/workflows/dotnet-core-win.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
 
     strategy:
       matrix:
-        version: [net452,net46,net461,netcoreapp3.1]
+        version: [net452,net46,net461,netcoreapp2.1,netcoreapp3.1]
```
Comment (cijothomas) [inline .github/workflows/dotnet-core-win.yml:15]: netcore2.1 is already installed on win image?

Comment (eddynaka) [reply .github/workflows/dotnet-core-win.yml:15]: yes, it is

File: .github/workflows/dotnet-core.yml [hunk 2]
```diff
@@ -19,10 +19,21 @@ jobs:
       with:
         fetch-depth: 0 # fetching all
 
-    - name: Setup .NET Core
+    - name: Setup .NET Core 2.1.807
```
Comment (cijothomas) [inline .github/workflows/dotnet-core.yml:22]: I think name can be just "Setup .NET Core 2.1" without mentioning the exact patch version.

Comment (eddynaka) [reply .github/workflows/dotnet-core.yml:22]: ok!

File: .github/workflows/dotnet-core-linux.yml [hunk 3]
```diff
@@ -12,15 +12,24 @@ jobs:
...
       with:
-        dotnet-version: 3.1.x
+        dotnet-version: 2.1.807
+
+    - name: Setup .NET Core 3.1.301
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 3.1.301
+    
+    - name: .net SxS
```
Comment (cijothomas) [inline .github/workflows/dotnet-core-linux.yml:30]: can you add comment to explain what this is?

File: .github/workflows/dotnet-core-linux.yml [hunk 4]
```diff
@@ -12,15 +12,24 @@ jobs:
...
     strategy:
       matrix:
-        version: [netcoreapp3.1]
+        version: [netcoreapp2.1,netcoreapp3.1]
 
     steps:
     - uses: actions/checkout@v2
 
-    - name: Setup .NET Core
+    - name: Setup .NET Core 2.1.807
```
Comment (cijothomas) [inline .github/workflows/dotnet-core-linux.yml:20]: simplify name by removing the .807 part.

File: .github/workflows/dotnet-core-linux.yml (hunk 3 above)
Comment (eddynaka) [reply .github/workflows/dotnet-core-linux.yml:30]: sure!

File: .github/workflows/dotnet-core-linux.yml (hunk 4 above)
Comment (eddynaka) [reply .github/workflows/dotnet-core-linux.yml:20]: i will do for all places.

File: .github/workflows/dotnet-core-linux.yml (hunk 3 above)
Comment (eddynaka) [reply .github/workflows/dotnet-core-linux.yml:31]: I added the following comment: `# For linux, we have to apply a workaround to enable both dotnet versions at same time`. We can get the reason in the link that i added in the pr description until the github team develops a way to maintain both of versions for dotnet at same time.

**Pull request merged**